### DISCUSSION TITLE
What's the reason behind requiring signed commits (real name/email policy)?

### DISCUSSION THREAD

**Discussion by adrian5** [containers/podman#17132] [Q&A]
I sometimes find issues in the docs, but unlike e.g. [systemd](https://github.com/systemd/systemd/) it was never possible to [contribute](https://github.com/containers/podman/discussions/10336) PRs without supplying a real user and email address, which I don't want to associate with my (mostly personal leisure) account. Does this requirement stem from the project's license or something else related to governance?

**Comment by vrothberg** [comment]
Thanks for reaching out!

You can find more information in the CONTRIBUTING file: https://github.com/containers/podman/blob/main/CONTRIBUTING.md#sign-your-prs

**Comment by adrian5** [comment]
But this is a hard requirement tied to the license used for podman, right? That's what I'm asking, i.e. why one cannot choose to "give away" one's contribution anonymously. Thanks.

**Reply by vrothberg** [reply]
I don't think it's tied to a specific license.  It's used beyond Apache 2.  To the best of my knowledge it's an extra layer of legal security to make sure projects cannot be put in danger by malinteded contributions.

> why one cannot choose to "give away" one's contribution anonymously

That is possible.  Some community members don't want to reveal their names for political reasons.  In these cases, we can open PRs if needed.

**Reply by Luap99** [reply]
As I understand the DCO was introduced in Linux, https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin
The DCO is what was lawyers found acceptable so many projects use it.

> That is possible. Some community members don't want to reveal their names for political reasons. In these cases, we can open PRs if needed.

I am not sure about this, it is best to ask the legal people first. In the end it boils down to copyright laws which very across countries.

**Reply by vrothberg** [reply]
> I am not sure about this, it is best to ask the legal people first. In the end it boils down to copyright laws which very across countries.

We did that already and have reached out to legal already.  I am not sure about copyright implications though.

**Reply by eslerm** [reply]
[DCO 1.1](https://developercertificate.org/) doesn't include the real name requirement itself. The *real name* requirement is appended afterwards in many projects like [GNU](https://gcc.gnu.org/dco.html) and previously the Linux Kernel. Many projects copied the DCO from Linux, which is why the *real name* clause is so prevalent today.

*Real name* is ambiguous and can mean legal name. [Linus recently changed the Kernel's DCO](https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?id=d4563201f33a022fc0353033d9dfeb1606a88330), since not allowing people to use their chosen (or common-law) name is discriminatory.
